source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -323,6 +323,12 @@ bool Filter::isBufferFull() const {
 }
 
 void Filter::continueDecoding() {
+  if (buffer_data_ && !skip_check_) {
+    // When the filter is asked to buffer the data but the buffer is full, it skips buffering more
+    // data for the next iteration.
+    buffer_data_ = !isBufferFull();
+  }
```

Comment:
Can you just do `buffer_data_ = false;` here? When we continue after a pause aren't we done and never want to buffer again?

Reply:
Yes, you're right. Thank you for this. Updated.